server/build.gradle

```diff
@@ -259,4 +259,7 @@ tasks.named('splitPackagesAudit').configure {
     // These are tricky because Lucene itself splits the index package,
     // but this should be fixed in Lucene 9
   ignoreClasses 'org.apache.lucene.index.LazySoftDeletesDirectoryReaderWrapper'
+
+  // only used temporarily, will be removed in a follow-up
+  ignoreClasses 'org.apache.lucene.index.SegmentInfosHelper'
```

Comment:
The SegmentInfosHelper class is used by this PR to load ES 6 (= Lucene 7) indices, which allows us to test that the recovery infrastructure as added by the PR works as intended. The class here will be removed in a follow-up step that will take a different, but more involved approach to load older indices (it's out of focus for this PR).